### PR TITLE
Include golangci-lint version in make target

### PR DESCRIPTION
**Pull request by g-gaston**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,6 +67,11 @@ KUSTOMIZE_OUTPUT_BIN_DIR="${OUTPUT_DIR}/kustomize-bin/"
 KUBEBUILDER := $(TOOLS_BIN_DIR)/kubebuilder
 KUBEBUILDER_VERSION := v3.1.0
 
+GOLANGCI_LINT_CONFIG ?= .github/workflows/golangci-lint.yml
+GOLANGCI_LINT_VERSION ?= $(shell cat $(GOLANGCI_LINT_CONFIG) | yq e '.jobs.golangci.steps[] | select(.name == "golangci-lint") .with.version' -)
```
Comment (lewisdiamond) [inline Makefile:71]: This should only be run if the file changes.

File: Makefile [hunk 2]
```diff
@@ -67,6 +67,11 @@ KUSTOMIZE_OUTPUT_BIN_DIR="${OUTPUT_DIR}/kustomize-bin/"
...
+GOLANGCI_LINT_DIR := $(TOOLS_BIN_DIR)
```
Comment (lewisdiamond) [inline Makefile:72]: I'd drop this variable and just use TOOLS_BIN_DIR now that you simplified it.

File: Makefile [hunk 3]
```diff
@@ -254,13 +259,11 @@ $(SETUP_ENVTEST): $(TOOLS_BIN_DIR)
...
-lint: bin/golangci-lint ## Run golangci-lint
-	bin/golangci-lint run
+lint: $(GOLANGCI_LINT) ## Run golangci-lint
+	$(GOLANGCI_LINT) run
 
-bin/golangci-lint: ## Download golangci-lint
-bin/golangci-lint: GOLANGCI_LINT_VERSION?=$(shell cat .github/workflows/golangci-lint.yml | yq e '.jobs.golangci.steps[] | select(.name == "golangci-lint") .with.version' -)
-bin/golangci-lint:
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s $(GOLANGCI_LINT_VERSION)
+$(GOLANGCI_LINT): $(GOLANGCI_LINT_CONFIG)
```
Comment (lewisdiamond) [inline Makefile:265]:
```suggestion
$(GOLANGCI_LINT): $(GOLANGCI_LINT_CONFIG)
    $(eval GOLANGCI_LINT_VERSION?=$(shell cat .github/workflows/golangci-lint.yml | yq e '.jobs.golangci.steps[] | select(.name == "golangci-lint") .with.version' -))
```

File: Makefile (hunk 1 above)
Comment (lewisdiamond) [inline Makefile:71]:
```suggestion
```

**Review by TerryHowe — APPROVED** [review]
/lgtm

nice! same results as from the action I suppose.

**Comment by TerryHowe** [comment]
I only put that hold on there because it was approved and lgtm and there were comments.

**Comment by g-gaston** [comment]
@TerryHowe @lewisdiamond comments addressed!

**Comment by lewisdiamond** [comment]
/lgtm
Feel free to unhold when you're ready.

**Pull request merged**